### PR TITLE
feat: sync v8.31.0

### PR DESCRIPTION
**Pull request by kecrily**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

sync v8.31.0 from eslint repo

### CODE REVIEW

**Review by mdjermanovic — COMMENTED** [review]
These changes look good.

Two questions:

1. Should we also include the fix in `formatters/index.md`? (https://github.com/eslint/eslint/commit/d9a39c7685496999b95b11486881d411089ab34e)
2. Since we're copying all files, the left sidebar menu now has duplicate items (e.g., there is [用户指南] and  Use ESLint in your project). Is that okay for now?

**Comment by kecrily** [comment]
> 1. Should we also include the fix in `formatters/index.md`? ([eslint/eslint@d9a39c7](https://github.com/eslint/eslint/commit/d9a39c7685496999b95b11486881d411089ab34e))

I will update it.

> 2. Since we're copying all files, the left sidebar menu now has duplicate items (e.g., there is [用户指南] and  Use ESLint in your project). Is that okay for now?

Structural changes are included in v8.32.0. I will finish it sync as soon as possible.

I think this PR should be merged together at the same time as the upcoming sync PR for v8.32.0 is approved. Before that this PR should be blocked.

**Comment by ollie-iterators** [comment]
This would be great to install because the chinese documentation has two versions of the links on the documentation site.
See: https://zh-hans.eslint.org/docs/latest/

**Comment by nzakas** [comment]
@kecrily are you saying we should merge #95 before this one or after it?

**Comment by kecrily** [comment]
> @kecrily are you saying we should merge #95 before this one or after it?

I originally thought that the current site did not deploy commits with structural changes, so needed to merge both v8.31.0 and v8.32.0 to avoid duplicate content.

But it turns out that I was wrong, and the current site has already deployed the commits that have changed structure, so in normal order, v8.31.0 is merged first and then v8.32.0 is merged.

But we should also merge v8.32.0 as soon as possible, because it's too bad that the menu bar has duplicate content.

----

I also plan to take steps to have the version of content automatically pulled match the translated version, rather than always pulling the latest docs.

**Comment by nzakas** [comment]
Makes sense!

**Review by nzakas — APPROVED** [review]
LGTM pending @mdjermanovic's review.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Comment by mdjermanovic** [comment]
Note that the HTML formatter page that should appear in an iframe is missing, but that doesn't seem to be caused by this particular change.

image

**Comment by mdjermanovic** [comment]
> I think this PR should be merged together at the same time as the upcoming sync PR for v8.32.0 is approved. Before that this PR should be blocked.

Makes sense, I'll let you merge this when and how you think it would be best.

File: src/rules/no-underscore-dangle.md
Comment (harish-sethuraman) [inline src/rules/no-underscore-dangle.md]:
```suggestion
* `"allowAfterThisConstructor": false`（默认值）不允许在 `this.constructor` 对象的成员中使用悬空的下划线
```
NIT Either remove `.` in all points or have it in all. Ideally better to remove

File: src/rules/no-underscore-dangle.md
Comment (harish-sethuraman) [inline src/rules/no-underscore-dangle.md]:
```suggestion
* `"enforceInClassFields": false`（默认值）允许在 es2022 类字段名中使用悬空的下划线
```

**Comment by kecrily** [comment]
Sorry it took so long to process, but my computer recently broke down and just got fixed.

**Pull request merged**
